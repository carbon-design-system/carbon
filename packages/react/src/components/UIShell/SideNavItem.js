/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { usePrefix } from '../../internal/usePrefix';

const SideNavItem = ({
  className: customClassName,
  children,
  large = false,
}) => {
  const prefix = usePrefix();
  const className = cx({
    [`${prefix}--side-nav__item`]: true,
    [`${prefix}--side-nav__item--large`]: large,
    [customClassName]: !!customClassName,
  });
  return <li className={className}>{children}</li>;
};

SideNavItem.propTypes = {
  /**
   * Provide a single icon as the child to `SideNavIcon` to render in the
   * container
   */
  children: PropTypes.node.isRequired,

  /**
   * Provide an optional class to be applied to the containing node
   */
  className: PropTypes.string,

  /**
   * Specify if this is a large variation of the SideNavItem
   */
  large: PropTypes.bool,
};

export default SideNavItem;
