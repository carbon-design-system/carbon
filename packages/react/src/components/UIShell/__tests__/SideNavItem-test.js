/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import SideNavItem from '../SideNavItem';

const prefix = 'cds';

describe('SideNavItem', () => {
  let mockProps, wrapper;

  beforeEach(() => {
    mockProps = {
      className: 'custom-classname',
      children: <span>foo</span>,
    };
  });

  afterEach(() => {
    wrapper && wrapper.unmount();
  });

  it('should render', () => {
    wrapper = mount(<SideNavItem {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should include a css class to render the large variant is large prop is set', () => {
    wrapper = mount(<SideNavItem {...mockProps} />);
    expect(
      wrapper.find('li').hasClass(`${prefix}--side-nav__item--large`)
    ).toBe(false);
    wrapper.setProps({ large: true });
    expect(
      wrapper.find('li').hasClass(`${prefix}--side-nav__item--large`)
    ).toBe(true);
  });
});
