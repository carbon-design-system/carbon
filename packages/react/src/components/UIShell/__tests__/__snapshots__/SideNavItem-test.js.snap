// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavItem should render 1`] = `
<SideNavItem
  className="custom-classname"
>
  <li
    className="cds--side-nav__item custom-classname"
  >
    <span>
      foo
    </span>
  </li>
</SideNavItem>
`;
